tm2/pkg/db/mem_db.go

```diff
@@ -229,7 +238,12 @@ func (itr *memDBIterator) Key() []byte {
 func (itr *memDBIterator) Value() []byte {
 	itr.assertIsValid()
 	key := []byte(itr.keys[itr.cur])
-	return itr.db.Get(key)
+	v, err := itr.db.Get(key)
+	if err != nil {
+		panic(err) // TODO(ajnavarro): refactor when changing Iterator interface to return errors when needed.
+	}
+
+	return v
```

Comment:
I would revert this and leave it for the next PR to add; just to avoid cluttering the code with TODOs